### DISCUSSION TITLE
Unable to spot error in schema

### DISCUSSION THREAD

**Discussion by sptrakesh** [danielaparker/jsoncons#324] [Q&A]
Hello,

I am trying out this library for schema validation, and ran into an issue with one of my unit tests. I found the following schema while searching for samples, and most online tools I tested it with, it seems to be valid. However, jsoncons rejects it with `Not a double`. error. Any idea where the error in the schema is?
```
{
    "$schema": "http://json-schema.org/draft-04/schema#",
    "description": "A product from Acme's catalog",
    "properties": {
      "id": {
        "description": "The unique identifier for a product",
        "type": "integer"
      },
      "name": {
        "description": "Name of the product",
        "type": "string"
      },
      "price": {
        "exclusiveMinimum": true,
        "minimum": 0,
        "type": "number"
      },
      "tags": {
        "items": {
          "type": "string"
        },
        "minItems": 1,
        "type": "array",
        "uniqueItems": true
      }
    },
    "required": ["id", "name", "price"],
    "title": "Product",
    "type": "object"
  }
```

**Comment by danielaparker** [comment]
Thanks for your feedback.

The current version of jsoncons only supports JSON Schema Draft 7. Support for later versions is planned, and possibly earlier versions depending on user interest. 

In Draft 4, exclusiveMinimum is a boolean, but in later drafts that was changed to a number (integer or floating point), hence the error.

Admittedly the error reporting when compiling the schema document is lacking. We should have checked the "$schema" keyword, and rejected draft-4 as not supported. And the error message "not a double" could have been more helpful, perhaps "exclusiveMinimum must be a number".  Those issues will be addressed in the next release.
